docs/for-developers/developer-overview.md

```diff
@@ -0,0 +1,213 @@
+
+For example, you may want to ban the use of `==` in JavaScript and instead require `===` to avoid **type coercion** when evaluating expressions, a common standard enforced in popular JavaScript linters. This is a simple find and replace in many text editors, because the ban is enforced for **all** usages of `==`. In Semgrep, you can create a rule codifying this find and replace operation to share or enforce this standard.
+
+<iframe title="Prevent type coercion in JavaScript ==" src="https://semgrep.dev/embed/editor?snippet=5rUdbO1" width="100%" height="432px" frameBorder="0"></iframe>
+_**Figure**. Prevent type coercion in `==`. Click **<i class="fa-solid fa-play"></i> Run** to view the findings._
+
+This simple rule is accurate because it only requires the syntax defined in `pattern` to match, not the semantics. The **metavariables** $A and $B always evaluate to some value on the left and right hand side of the `==` operator, and that is all that matters, not the meaning or of $A and $B themselves.
+
+:::info Metavariables
+[Metavariables](/writing-rules/pattern-syntax#metavariables) are an abstraction to match code when you don’t know the value or contents ahead of time, similar to capture groups in regular expressions.
+:::
+</details>
+
+<details>
+<summary>Complex syntax-based example: ban `console.log` in external or user-facing functions</summary>
+
+#### Complex syntax-based example
+
+It is a common convention to ban all uses of some language feature in user-facing code, such as `console.log()`, or `console.log()` may be permitted internally but not externally.
+
+Semgrep enables you to create a custom best practices set of rules around cases like this.
+
+<iframe title="Ban console.log external or user-facing functions" src="https://semgrep.dev/embed/editor?snippet=1AP5" width="100%" height="432px" frameBorder="0"></iframe>
+_**Figure**. Ban `console.log` in external-facing functions. Click **<i class="fa-solid fa-play"></i> Run** to view the findings._
+
+Notice that only **line 4** matches. This is because only line 4 has a `console.log()` function within `someExternalFunction()`.
+
+This example defines both what matches within the external-facing function, and the external-facing function itself. This is achieved through the use of `pattern` and `pattern-inside`. The `...` **ellipsis** operator tells Semgrep to accept any number of arguments or values in `someExternalFunction()` and `console.log()`, thus capturing all possible variations of the functions.
+
+
+</details>
+
+<details>
+<summary>Semantic taint analysis: detecting unsanitized data from source to sink</summary>
+
+#### Semantic taint analysis example
+
+A more complex example is detecting if **unsanitized data** is flowing from some **source**, such as saved form data, to a **sink** without sanitization.
+
+The following example is a simplified Semgrep rule that detects possible cross-site scripting vulnerabilities:
+
+
+```yaml showLineNumbers
```

Comment:
This 3rd example doesn't have an embedditor (it requires log in to view the rule). How do we feel about that? I could also convert the other two previous examples into plain YAML / source code rather than use the embedditor.

Reply:
:ACK: